sdks/nodejs/spec/agonesSDK.spec.js

```diff
@@ -187,6 +187,49 @@ describe('AgonesSDK', () => {
 			}
 		});
 
+		it('calls the server and handles stream write error if callback provided', async () => {
+			let stream = jasmine.createSpyObj('stream', ['write', 'on']);
+			stream.write.and.callFake((chunk, encoding, callback) => {
+				callback('error');
+			});
+			spyOn(agonesSDK.client, 'health').and.callFake(() => {
+				return stream;
+			});
+			try {
+				agonesSDK.health(() => {});

```

Comment:
It may be worth adding an expect in here to test/demonstrate that the error is passed to the callback intact.
Could spy on the callback and I think and check what it was called with